src/processor/commit_state.rs

```diff
@@ -126,17 +126,14 @@ pub(crate) fn process_commit_state_internal(
     let mut delegation_metadata =
         DelegationMetadata::try_from_bytes_with_discriminator(&delegation_metadata_data)?;
 
-    // If the commit slot is greater or equal than the last update slot, we can proceed.
-    // If the slot is less, we simply do not commit.
-    // Since commit instructions are typically bundled, we return without error
-    // so that correct commits are executed.
-    if args.commit_record_slot < delegation_metadata.last_update_external_slot {
+    // To preserve correct history of account updates we require sequential commits
+    if args.commit_record_slot != delegation_metadata.last_update_external_slot + 1 {
         msg!(
-            "Slot {} is outdated, previous slot is {}. Skipping commit",
+            "Slot {} is outdated, previous slot is {}. Rejecting commit",
             args.commit_record_slot,
             delegation_metadata.last_update_external_slot
         );
-        return Ok(());
+        return Err(DlpError::OutdatedSlot.into());
```

Comment:
This change needs to be evaluated carefully with the new logic of the committer service, since before invalid commits were skipped, without failing the instruction. This means that if we are committing A, B, C in a single tx and B in invalid, the commit would fail for all accounts. This is probably correct and intended with the new logic, but flagging to double check